vsintegration/src/FSharp.Editor/CodeLens/AbstractCodeLensDisplayService.fs

```diff
@@ -171,20 +176,32 @@ type CodeLensDisplayService (view : IWpfTextView, buffer : ITextBuffer, layerNam
             self.UiElements.Remove trackingSpan |> ignore
             try
                 self.CodeLensLayer.RemoveAdornment(Grid) 
-            with e -> 
+            with e ->
+#if DEBUG
                 logExceptionWithContext(e, "Removing line lens")
+#endif
+                e |> ignore
+                ()
```

Comment:
Unit is not really needed here.


what is the purpose of e >ignore   ?

Reply:
I assume it's to make compiler happy. Otherwise you won't use e in that #ifdef branch and therefore unused variable detection would cry

Reply:
But more general question is: why do we just suck the exception and ignore it? Shouldn't we do something about it?

Reply:
Correct, this makes the compiler happy because our build sets warnings as errors.

> why do we just suck the exception and ignore it? 

My main goal here is to stop all logging unless we are in a debug context. One of the complaints about this when it's turned on is that it logs copious amounts of information that brings the F# Language Service Window to the foreground each time.

Reply:
name it ````_e````
or write it as ```` ignore e ```` then the e |>ignore makes me think the side effects, of which there can be none are important.

Reply:
`_e` won't be enough; we'll either need to ignore it or do `()`

I agree that having both is extraneous